modules/rostests/apitests/crt/mbtowc.c

```diff
@@ -0,0 +1,46 @@
+/*
+ * PROJECT:         ReactOS API tests
+ * LICENSE:         GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:         Tests for mbtowc
+ * COPYRIGHT:       Copyright 2020 Bișoc George <fraizeraust99 at gmail dot com>
+ */
+
+#include <apitest.h>
+
+#define WIN32_NO_STATUS
+#include <stdio.h>
+#include <stdlib.h>
+
+#define ok_wchar(x, y) ok_int(x, y)
+
+START_TEST(mbtowc)
+{
+    size_t Length;
+    wchar_t BufferDest[3];
+    char ch = 'A';
+
+    /* Everything is NULL */
+    Length = mbtowc(NULL, NULL, 0);
+    ok(Length == 0, "Expected 0 characters to be converted as everything is NULL but got %u.\n", Length);
+
+    /* Don't examine the number of bytes pointed by multibyte parameter */
+    Length = mbtowc(BufferDest, &ch, 0);
+    ok(Length == 0, "Expected 0 characters to be converted but got %u.\n", Length);
+
+    /* Wide character argument is invalid */
+    Length = mbtowc(NULL, &ch, 0);
+    ok(Length == 0, "Expected 0 characters to be converted but got %u.\n", Length);
+
+    /* The multibyte argument is invalid */
+    Length = mbtowc(BufferDest, NULL, 0);
+    ok(Length == 0, "Expected 0 characters to be converted but got %u.\n", Length);
+
+    /* The multibyte argument is invalid but count number for examination is correct */
+    Length = mbtowc(BufferDest, NULL, MB_CUR_MAX);
+    ok(Length == 0, "Expected 0 characters to be converted but got %u.\n", Length);
+
+    /* Convert the character and validate the output that we should get */
+    Length = mbtowc(BufferDest, &ch, MB_CUR_MAX);
```

Comment:
Another useful test would be this with a NULL wchar parameter, to show that you can get the length without receiving output.